### PR TITLE
Changed the old build system to the new: gradle.

### PR DESCRIPTION
**Pull request by raulccabreu**
Also changed the sample project to gradle with SwitchBoard android library as a
gradle module with path specified in settings.gradle

Project structures based on newest Android Studio Empty Activity Template.
(changing all activities extends to AppCompatActivity)

One needed additional adjustment (needed because network run on ui thread exception):

The sample SwitchBoard.loadConfig on activities OnCreate moved to
a Thread inside onStart, and applying the AB config changes after the
config load inside runOnUiThread.

This not satisfy the KeepSafe/Switchboard#20 but it's on way.

### CODE REVIEW

**Comment by raulccabreu** [comment]
We have this other PR Catbag/Switchboard#2 that add on android client a maven artifact upload to https://bintray.com/ and can be used on android apps as: ```compile 'br.com.catbag.switchboard:android-client:1.0.0'``` like KeepSafe/Switchboard#20 asks. If this pull request KeepSafe/Switchboard#22 is accepted i do then the next related PR.

**Comment by philippb** [comment]
Hi,
thank you for the PR. Sorry I didn't see that before. Let me look into it.

**Comment by philippb** [comment]
I like the idea of updating the example project to gradle. The `AppCompatActivity` adds unnecessary complexity to the sample app. The goal is to show in the most simple way on how this works and be able to build it. 
AppCombatActivity leads to `compile 'com.android.support:appcompat-v7:23.4.0'` as dependency that is not needed.
Also I don't like the external dependency to the switchboard artifact. 

Ideally you create a simple PR that just switches to gradle in the simplest form. I think the initiation of Switchboard should just happen in an `Application.onCreate()` on a background thread.

I will be working on cleaning some things up in here and make it an official artifact that can be linked as a dependency. 

If you have additional suggestions on API improvements, please open a new issue. I'm keen on learning what your experience is.

**Comment by raulccabreu** [comment]
@philippb, I did not understand this part: 

> Also I don't like the external dependency to the switchboard artifact.

Did you mean gradle local path lib module reference on example?

File: client/android/example/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
+    testCompile 'junit:junit:4.12'
+    compile project(path: ':lib')
+    compile 'com.android.support:appcompat-v7:23.4.0'
```
Comment (philippb) [inline client/android/example/app/build.gradle]: why is this a dependency? We should not need this - as it's just basic activities and switchboard.

File: client/android/example/app/src/main/java/com/keepsafe/switchboard/example/NextActivityNormal.java [hunk 2]
```diff
@@ -15,14 +15,13 @@
 */
 package com.keepsafe.switchboard.example;
 
-import android.app.Activity;
 import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
 
-public class NextActivityNormal extends Activity {
+public class NextActivityNormal extends AppCompatActivity {
```
Comment (philippb) [inline client/android/example/app/src/main/java/com/keepsafe/switchboard/example/NextActivityNormal.java]: make this a normal activity so we can drop the dependency. As it's a sample for switchboard `Activity` should be enough

**Comment by philippb** [comment]
your question: I think I misread that in the `gradle.build` file there was a dependency to your artifactory.
I can't see it anymore.

**Comment by raulccabreu** [comment]
@philippb Changes made. I will make a squash rebase when you approve.

**Comment by raulccabreu** [comment]
@philippb Hello?

**Comment by philippb** [comment]
Sorry been busy. Will look into it maybe today or next week. Thank you

**Comment by philippb** [comment]
Merged. Thank you very much for your work.

**Pull request merged**